### ISSUE TITLE
Incorrect bounding rects returned from pygame.draw.line/aaline

### ISSUE THREAD

**Issue by charlesej** [pygame/pygame#895]
The bounding rects returned from `pygame.draw.line` and `pygame.draw.aaline` are incorrect in some instances.

The related [pygame.draw documentation](https://www.pygame.org/docs/ref/draw.html) states (with the assumption that 'bounding area' means 'minimum bounding rectangle'):
> The functions return a rectangle representing the bounding area of changed pixels.

Below are a few examples of some issues with the returned bounding rects.

---

The following code draws a line (white) and its returned bounding rect (red).
```
import pygame
import time

pygame.init()
screen = pygame.display.set_mode((320, 320))

rect = pygame.draw.line(screen, pygame.Color('white'), (-1, -1), (320, 320), 1)
pygame.draw.rect(screen, pygame.Color('red'), rect, 1)
print(rect)

pygame.display.flip()
time.sleep(2)
pygame.quit()
```

The above code gives the following output. The returned bounding rect is outside of the screen's bounds. It should consist of coordinates that can be drawn on the screen.
```
pygame 1.9.5.dev0
Hello from the pygame community. https://www.pygame.org/contribute.html
<rect(-1, -1, 322, 322)>
```

![draw_line_bounding_rect](https://user-images.githubusercontent.com/4837704/54077716-7f4a3300-428a-11e9-9398-d90c3f763f29.png)

---

The following code draws a thick line (white) and its returned bounding rect (red).
```
import pygame
import time

pygame.init()
screen = pygame.display.set_mode((320, 320))

rect = pygame.draw.line(screen, pygame.Color('white'), (60, 240), (240, 60), 30)
pygame.draw.rect(screen, pygame.Color('red'), rect, 1)
print(rect)

pygame.display.flip()
time.sleep(2)
pygame.quit()
```

The above code gives the following output. The returned bounding rect is incorrect. It should be just big enough to fully enclose the line with no parts of the line outside of it.
```
pygame 1.9.5.dev0
Hello from the pygame community. https://www.pygame.org/contribute.html
<rect(60, 60, 210, 181)>
```

![draw_thickline_bounding_rect](https://user-images.githubusercontent.com/4837704/54077718-840ee700-428a-11e9-84c7-b72b216b99ea.png)

---

The following code draws an aaline (white) and its returned bounding rect (red).
```
import pygame
import time

pygame.init()
screen = pygame.display.set_mode((320, 320))

rect = pygame.draw.aaline(screen, pygame.Color('white'), (-1, -1), (320, 320))
pygame.draw.rect(screen, pygame.Color('red'), rect, 1)
print(rect)

pygame.display.flip()
time.sleep(2)
pygame.quit()
```

The above code gives the following output. Part of the returned bounding rect is outside of the screen's bounds. It should consist of coordinates that can be drawn on the screen.
```
pygame 1.9.5.dev0
Hello from the pygame community. https://www.pygame.org/contribute.html
<rect(0, 0, 321, 321)>
```

![draw_aaline_bounding_rect](https://user-images.githubusercontent.com/4837704/54077720-8b35f500-428a-11e9-8a9b-91dd7ba1c7d2.png)

---

System details:
- os: windows 10 (64bit)
- python: 3.7.2 (64bit)
- pygame: 1.9.5.dev0 (SDL: 1.2.15) at 811eaf051e86a4c711d42a0d426dfc138668f384

**Comment by charlesej** [comment]
The code for `draw.line` and `draw.aaline` is separate so this issue can be done in 2 parts. Tests for each will be added via issue #896.
- [ ] Fix bounding rects for `pygame.draw.line`
- [ ] Fix bounding rects for `pygame.draw.aaline`

**Comment by charlesej** [comment]
Test added for `pygame.draw.aaline` with `@unittest.expectedFailure` decorator via #1170.

**Comment by charlesej** [comment]
Resolved by #1494.
